### PR TITLE
feat(TextField): add a controlled variant

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/6887647837

### CODE REVIEW

File: packages/core/src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -171,13 +176,25 @@ const TextField: VibeComponent<TextFieldProps, unknown> & {
       [onChange, isRequiredAndEmpty]
     );
 
-    const { inputValue, onEventChanged, clearValue } = useDebounceEvent({
+    const {
+      inputValue: uncontrolledInput,
```
Comment (YossiSaadi) [inline packages/core/src/components/TextField/TextField.tsx:180]: why?

**Review by YossiSaadi — COMMENTED** [review]
I can't understand why if something is controlled it shouldn't get the logic of debounce, etc.

File: packages/core/src/components/TextField/TextField.tsx (hunk 1 above)
Comment (talkor) [reply packages/core/src/components/TextField/TextField.tsx:180]: The thing is that the debounce has a state, this is the root cause of the issue. I can add debounce but I think that if the user wants to control the input they should have the full flexibility

File: packages/core/src/components/TextField/TextField.tsx [hunk 2]
```diff
@@ -171,13 +176,25 @@ const TextField: VibeComponent<TextFieldProps, unknown> & {
...
       trim
     });
 
+    const inputValue = useMemo(() => {
+      return controlled ? value : uncontrolledInput;
+    }, [controlled, value, uncontrolledInput]);
+
+    function handleChange(event: ChangeEvent<Partial<HTMLInputElement>>) {
+      controlled ? onChangeCallback(event.target.value) : onEventChanged(event);
+    }
```
Comment (YossiSaadi) [inline packages/core/src/components/TextField/TextField.tsx:196]:
```suggestion
const handleChange = useCallback<ChangeEventHandler>(event => {
      // ...
    }, []);
```

**Review by YossiSaadi — APPROVED** [review]
We need, when instructing people to use this new `controlled` to let them know they lose some of the built in functionality as debounce

**Pull request merged**